alert-common/src/main/java/com/synopsys/integration/alert/common/persistence/model/UserModel.java

```diff
@@ -67,17 +69,22 @@ public String getEmailAddress() {
         return emailAddress;
     }
 
-    public Collection<String> getRoles() {
+    public Collection<UserRoleModel> getRoles() {
         return roles;
     }
 
     public Collection<GrantedAuthority> getRoleAuthorities() {
         // Spring requires the roles to start with ROLE_
-        return roles.stream().map(role -> ROLE_PREFIX + role).map(SimpleGrantedAuthority::new).collect(Collectors.toList());
+        return roles.stream()
+                   .map(UserRoleModel::getName)
+                   .map(role -> ROLE_PREFIX + role)
+                   .map(SimpleGrantedAuthority::new)
+                   .collect(Collectors.toList());
     }
 
     public boolean hasRole(final String role) {
-        return roles.contains(role);
+        //Make this performs better.
```

Comment:
If this is the solution for now, add a TODO so we don't lose track of it. Also: "performs" -> "perform".